src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -816,7 +820,11 @@ def create_arm_client_factory(credentials):
             from azure.cli.core.profiles._shared import get_client_class
             from azure.cli.core.profiles import ResourceType, get_api_version
             from azure.cli.core.commands.client_factory import configure_common_settings
+            from azure.cli.core.azclierror import CLIInternalError
             client_type = get_client_class(ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS)
+            if client_type is None:
+                raise CLIInternalError("Unable to get '{}' in profile '{}'"
+                                       .format(ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS, cli_ctx.cloud.profile))
```

Comment:
For some weird cases(CX upgrade or degrade MGMT_RESOURCE_SUBSCRIPTIONS themselves, I guess.), `get_client_class` will fail and return None. If not verified, line 829 will throw `TypeError: _NoneType_ object is not callable`

Reply:
I am ok with it, currently.

But we may need go deeper to see what are the exact reasons that cause the errrors and provide the corresponding suggestions. Because our goal is to resolve all the CLIInternalErrors

Reply:
This is because `get_client_class` returned `None`. In such case `azure-mgmt-resource` SDK is somehow missing. 

`get_client_class` should actually raise an `Exception` instead of returning `None`. This is a design flaw and need further discussion.